dafni_cli/datasets/dataset_metadata.py

```diff
@@ -1,181 +1,371 @@
+    """Dataclass representing the standard listed in a dataset's metadata
+
+    Attributes:
+        type (str): Standard type
+        id (str or None): Standard id
+        label (str or None): Standard label
+    """
 
+    type: str
+    id: Optional[str] = None
+    label: Optional[str] = None
 
-class DatasetMetadata:
-    """Class to represent the DAFNI Dataset
-    Metadata client model
+    _parser_params: ClassVar[List[ParserParam]] = [
+        ParserParam("id", "@id", str),
+        ParserParam("type", "@type", str),
+        ParserParam("label", "label", str),
+    ]
+
+    def __str__(self) -> str:
+        """Nicer string representation for printing"""
+        return f"{self.label}"
+
+
+@dataclass
+class DatasetMetadataVersion(ParserBaseObject):
+    """Dataclass representing a metadata version listed in a dataset's
+    version_history
+
+    Attributes:
+        metadata_id (str): ID of this version of the metadata
+        label (str): Label of the metadata version
+        modified_date (datetime): Time and date of last modification of this
+                                  version
+    """
+
+    metadata_id: str
+    label: str
+    modified_date: datetime
+
+    _parser_params: ClassVar[List[ParserParam]] = [
+        ParserParam("metadata_id", "metadata_uuid", str),
+        ParserParam("label", "dafni_version_note", str),
+        ParserParam("modified_date", "modified_date", parse_datetime),
+    ]
+
+
+@dataclass
+class DatasetVersion(ParserBaseObject):
+    """Dataclass representing a specific dataset version listed under a
+       dataset's metadata version_history
+
+    Attributes:
+        version_id (str): ID of this version of the dataset
+        metadata_versions (List[DatasetMetadataVersion): List of metadata
+                              versions under this version of the dataset
+    """
+
+    version_id: str
+    metadata_versions: List[DatasetMetadataVersion]
+
+    _parser_params: ClassVar[List[ParserParam]] = [
+        ParserParam("version_id", "version_uuid", str),
+        ParserParam("metadata_versions", "metadata_versions", DatasetMetadataVersion),
+    ]
+
+
+@dataclass
+class DatasetVersionHistory(ParserBaseObject):
+    """Dataclass for processing the version history of a dataset
+
+    Methods:
+        process_version_history(session: DAFNISession, json_flag: bool): Iterates through all versions and outputs details
+
+    Attributes:
+        session (DAFNISession): User session
+        dataset_id (str): Dataset ID
+        versions (List[dict]): List of associated Version dicts
+        version_ids (List[str]): List of Version IDs
+    """
+
+    dataset_id: str
+    versions: List[DatasetVersion]
+
+    _parser_params: ClassVar[List[ParserParam]] = [
+        ParserParam("dataset_id", "dataset_uuid", str),
+        ParserParam("versions", "versions", DatasetVersion),
+    ]
+
+    def process_version_history(self, session: DAFNISession, json_flag: bool = False):
```

Comment:
Function name - maybe change to something like `process_and_print_version_history`